src/app/client/src/app/modules/learn/components/batch/batch-details/batch-details.component.html

```diff
@@ -1,102 +1,133 @@
-                            class="ui right floated sb-btn sb-btn-normal" appTelemetryInteract
-                            [telemetryInteractEdata]="enrollBatchIntractEdata"
-                            [telemetryInteractObject]="{id:batch.identifier,type:'CourseBatch',ver:'1',rollup:{l1:courseId}}"
-                            (click)="enrollBatch(batch)"
-                            *ngIf="!courseMentor">{{resourceService.frmelmnts.btn.enroll}}</button>
-                        <span class="header pb-5"> {{batch.startDate | dateFormat}} - </span>
-                        <span *ngIf="batch.endDate" class="header pb-5"> {{batch.endDate | dateFormat}} </span>
-                        <div class="description" *ngIf="userNames[batch.createdBy]">Created
-                            by {{userNames[batch.createdBy].firstName}} {{userNames[batch.createdBy].lastName}}</div>
-                        <span *ngIf="isValidEnrollmentEndDate(batch.enrollmentEndDate)">
-                            <span class="sb-label-12size"
-                                *ngIf="!isEnrollmentAllowed(batch.enrollmentEndDate)">{{resourceService.messages.stmsg.m0136}}
-                                {{batch.enrollmentEndDate | dateFormat}}</span>
-                            <span class="sb-label-status-error"
-                                *ngIf="isEnrollmentAllowed(batch.enrollmentEndDate)">{{resourceService.messages.stmsg.m0134}}</span>
-                        </span>
-                    </div>
-                </div>
-                <div class="content my-15" *ngIf="batchList.length <= 0">
-                    <div class="sb-no-batch-found">{{resourceService.messages.stmsg.m0081}}</div>
-                </div>
+    <div class="sb-modal-content" *ngIf="showAllBatchList">
+        <div class="join-batch">
+            <div class="join-batch__heading">
+                <div>{{resourceService?.frmelmnts?.lbl?.availableBatches}}</div>
+                <div>{{resourceService?.frmelmnts?.lbl?.courseCreatedBy}}</div>
             </div>
-            <div *ngIf="showError">
-                <div class="sb-no-batch-found">{{resourceService.messages.stmsg.m0081}}</div>
+            <div class="join-batch__body" *ngFor="let batch of allBatchList">
+                <div>{{batch.startDate | dateFormat}}<span *ngIf="batch.endDate"> - {{batch.endDate | dateFormat}}
+                    </span></div>
+                <div *ngIf="userNames[batch.createdBy]">{{userNames[batch.createdBy].firstName}}
+                    {{userNames[batch.createdBy].lastName}}</div>
+                <button [disabled]="isEnrollmentAllowed(batch?.enrollmentEndDate)"
+                    [ngClass]="{'sb-btn-disabled': isEnrollmentAllowed(batch?.enrollmentEndDate), 'sb-btn-primary': !isEnrollmentAllowed(batch.enrollmentEndDate)}"
+                    class="sb-btn sb-btn-outline-secondary sb-btn-normal" appTelemetryInteract
+                    [telemetryInteractEdata]="enrollBatchIntractEdata"
```

Comment:
This should have cdata which should include the course

Reply:
Done